x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/SearchableSnapshotAction.java

```diff
@@ -78,7 +78,11 @@ public class SearchableSnapshotAction implements LifecycleAction {
         indexName,
         state) -> state.forceMergeCloneIndexName() != null ? state.forceMergeCloneIndexName() : indexName;
 
-    private static final Settings CLONE_SETTINGS = Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0).build();
+    /** The cloned index should have 0 replicas, so we also need to remove the auto_expand_replicas setting if present. */
+    private static final Settings CLONE_SETTINGS = Settings.builder()
+        .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+        .put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, (String) null)
```

Comment:
hmm, isn't `false` the right thing here?



seems like this will brick based on my reading of: https://github.com/szybia/elasticsearch/blob/680e7a697995aa08ebe881a1e23e710da891291b/server/src/main/java/org/elasticsearch/cluster/metadata/AutoExpandReplicas.java#L54

what am i missing?
```suggestion
        .put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, "false")
```

Reply:
Sorry, I misread before. The default is `false`: https://github.com/elastic/elasticsearch/blob/680e7a697995aa08ebe881a1e23e710da891291b/server/src/main/java/org/elasticsearch/cluster/metadata/AutoExpandReplicas.java#L40-L46
That does avoid having replicas. And the default `null` logic is built into the settings infrastructure. I don't know off the top of my head where that lives, but I can try to find it if you're curious.

Reply:
Oh sorry yeah I meant false!